automation/taskcluster/lib/util.py

```diff
@@ -0,0 +1,7 @@
+import re
+
+
+def convert_camel_case_into_kebab_case(string):
```

Comment:
Nit: Perhaps this function should be moved into `tasks.py` since it's the only place that it's needed?
Then, at the time where other files start to need this behaviour, we are forced to decide whether we want to put it into a `utils` file, or (since we have the additional context of that new usage) we can realize which part of the domain this is associated with, and structure our code accordingly - e.g.: perhaps this code is only used with variant code, so would be placed in the same location.

I lean a lot on the idea of ["YAGNI"](https://martinfowler.com/bliki/Yagni.html) until code is needed (or, in this case, until code needs to be shared between files), remove it (or scope it as privately as possible).

-----

Just a nit, not blocking this PR. Let me know what you think :)

Reply:
I see it's the same argument as https://github.com/mozilla-mobile/fenix/pull/1220#discussion_r270051072 where `populate_*` functions are only used in decision_task.py. I don't think this function itself is related to tasks, though, that's why I think it's better in its own module. Anyway, I don't mind moving it there. I don't it's worth, though, because we're eventually moving to taskgraph.

Reply:
Yeah, it's a very similar argument to that other PR :smile:

> I don't think this function itself is related to tasks, though

Perhaps it's not related to tasks, specifically, but it _is_ related to variants and indexes (and converting parts of the variant to be compatible with the index). In our (relatively-small) automation-scheduling system here in `r-b`, we currently have all logic for variants and indexes here in `tasks.py`. Therefore, transitively, our variant-/index-specific logic should be in `tasks.py`.

-----

To cover my bases here: there's the argument that, perhaps, converting camel-to-kebab isn't variant-related. Since other code may theoretically want to convert strings from camel-to-kebab, shouldn't this logic be shared? In doing so we can avoid duplicating code (DRY)!

My counterargument here would be the difference between [code duplication and behaviour duplication](https://anthonysciamanna.com/2018/07/28/the-dry-principle-and-incidental-duplication.html). If two independent parts of code need to convert from camel-to-kebab (and they aren't both variant-related logic), then they aren't sharing behaviour: they're just sharing code!
You can confirm this assumption with the question: if Taskcluster indexes changed so they didn't allow dashes, and you had to use underscores instead, could you safely change `convert_camel_case_into_kebab_case` without breaking the other (theoretical) usage?

To illustrate this, you can see how `convert_camel_case_into_kebab_case` could be renamed (and mildly modified) to the domain-specific function `convert_variant_to_index_postfix(variant) -> '$product.$arch-$build_type`. At that point, it's now domain-specific, and it's more clear how it pertains specifically to variants, and therefore should be moved to be beside the rest of the variant logic

Reply:
I'm sorry, I think this conversation is turning into bike-shedding. This function does simple with any strings, variants or not. Let's keep it this way, for now and see how it evolves. The stability of `automation/taskcluster/` is not at risk if we leave it in a separate module, for now.